src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -316,11 +325,18 @@ public function updateAction(Request $request)
                 return $this->redirectHandler->redirectToResource($configuration, $resource);
             }
 
-            if ($configuration->hasStateMachine()) {
-                $this->stateMachine->apply($configuration, $resource);
+            try {
+                $this->resourceUpdater->applyTransitionAndFlush($resource, $configuration, $this->manager);
+            } catch (RaceConditionException $exception) {
+                if (!$configuration->isHtmlRequest()) {
+                    return $this->viewHandler->handle($configuration, View::create($form, Response::HTTP_BAD_REQUEST));
+                }
+
+                $this->flashHelper->addErrorFlash($configuration, 'race_condition_error');
```

Comment:
Or even could be configurable as the api code (https://github.com/Sylius/Sylius/pull/7425#discussion_r101492048)